app/scripts/controllers/transactions/index.js

```diff
@@ -331,6 +332,216 @@ export default class TransactionController extends EventEmitter {
     });
   }
 
```

Comment:
Hey @segun , I think we should also capture transaction metrics in update function.

Reply:
@danjm see my latest push. I added some specific updates already. Let me know if I should leave them in or remove them.

Reply:
I agree with @danjm, we probably do not need event capturing at this stage.

Reply:
We probably should capture fields in some of events like gas fees changing. I see bit of issue in how metrics have been implemented in this PR.

We should not be adding `TRANSACTION_EVENTS` only capturing new fields. We can just call `_createTransactionEventFragment` to add new fields for transaction metrics.

cc @brad-decker can you plz suggest.

Reply:
As agreed, I'll leave out metrics from this PR>